### ISSUE TITLE
Rules fail with ESLint 9.0.0 - `context.getScope is not a function`

### ISSUE THREAD

**Issue by tillig** [lo1tuma/eslint-plugin-mocha#352]
It appears there have been quite a few breaking changes with ESLint 9.0.0, and with that it looks like `context.getScope()` is no longer an available shortcut.

- It was [there in v8.57.0 of ESLint](https://github.com/eslint/eslint/blob/abea3b6f399fde9e28cf6dc5d6c013938f163cdc/lib/linter/linter.js#L986)
- It's [gone in v9.0.0 of ESLint](https://github.com/eslint/eslint/blob/e0cbc50179adac1670f4e0bd9093387a51f4f42a/lib/linter/linter.js#L1040)

It looks like [a PR got merged that has the fix for this](https://github.com/lo1tuma/eslint-plugin-mocha/pull/347) back on Feb 16, but it appears that hasn't been released yet - it's not in v10.4.1, which is the current latest.

Is there a plan to release that in the near future?

**Comment by MikeMcC399** [comment]
This issue relates to the blog [Preparing your custom rules for ESLint v9.0.0](https://eslint.org/blog/2023/09/preparing-custom-rules-eslint-v9/) published on Sep 26, 2023 which has the comment:

> All of the context methods listed in this table will be removed in ESLint v9.0.0, and the replacement methods on SourceCode have already been in place for six years, so you should have no problem switching to the new methods. (Yes, we deprecated these and then completely forgot to remove them.)

Good to see that there is already a merged PR for this as [Eslint v9](https://eslint.org/blog/2024/04/eslint-v9.0.0-released/) was announced released on Apr 5, 2024.

**Comment by lo1tuma** [comment]
Thanks for reporting. I will take a look into it and try to make a patch release ASAP.

**Comment by lo1tuma** [comment]
I’ve cherry-picked the changes from #347 and made a patch release v10.4.2. Unfortunately I forgot to make a release after merging #347 to `main` and now `main` contains a few breaking changes which are planned for v11.

Please let me know if v10.4.2 fixes the issues with ESLint v9.

**Comment by lo1tuma** [comment]
@MikeMcC399 thanks for confirming... in this case I will close this issue.

**Comment by tillig** [comment]
Yup, also verified. Thanks!

**Comment by mcat95** [comment]
Hello,

I believe the 10.4.2 patch broke compatibility with eslint versions prior to 8 (We get the error on 7.32.0):
`TypeError: Cannot read properties of undefined (reading 'getScope')`

**Comment by MikeMcC399** [comment]
@mcat95 

> I believe the 10.4.2 patch broke compatibility with eslint versions prior to 8 (We get the error on 7.32.0): `TypeError: Cannot read properties of undefined (reading 'getScope')`

Ouch!

It sounds like this needs to be a new issue leading to a new fix release.

The example in https://eslint.org/blog/2023/09/preparing-custom-rules-eslint-v9/#context.getscope() shows supporting both the deprecated API and the new API. If only the new API has been implemented, then it would break compatibility with anything below ESLint `v8.37.0` 🤭 

```js
module.exports = {
    create(context) {

        const { sourceCode } = context;

        return {
            Program(node) {
                const scope = sourceCode.getScope
                    ? sourceCode.getScope(node)
                    : context.getScope();

                // do something with scope
            }
        }
    }
};

```

**Comment by MikeMcC399** [comment]
@mcat95 

- It's good that you spotted this issue and reported it! I've submitted it as a new open issue #354.

**Comment by MikeMcC399** [comment]
@mcat95 

This was fixed in [10.4.3](https://github.com/lo1tuma/eslint-plugin-mocha/releases/tag/10.4.3).
